app.js

```diff
@@ -1,30 +1,133 @@
 import express from "express";
 import nunjucks from "nunjucks";
-import axios from "axios";
+import mongoose from "mongoose";
+import * as uuid from "uuid";
+import dotenv from "dotenv";
+import fs from "fs";
+import filterGameOptions from "./server/js/filterGameOptions.js";
+import wordCheck from "./server/js/wordCheck.js";
+import Game from "./server/models/Game.js";
+import HighScore from "./server/models/highScore.js";
+import saveInDb from "./server/js/saveInDB.js";
+import getOneFromDB from "./server/js/getOneFromDB.js";
+import updateOneFromDB from "./server/js/updateOneFromDB.js"
+import highScoreRouter from "./server/js/highScoreRouter.js";
+
+dotenv.config();
 
 const app = express();
 
-const env = nunjucks.configure('views', {
-    autoescape: true,
-    express: app
+const DBUrl = process.env.DB_URL;
+
+const env = nunjucks.configure("views", {
+  autoescape: true,
+  express: app,
 });
 
-app.set('view engine', 'html');
+app.set("view engine", "html");
 
 app.use(express.static("./public"));
 
-app.get("/", (req,res)=> {
-    res.render("index.html");
+app.use(express.json());
+
+app.use("/high-scores", highScoreRouter)
+
+app.get("/", (req, res) => {
+  res.render("index.html");
 });
 
-app.get("/information", (req,res)=> {
-    res.render("information.html");
+app.get("/information", (req, res) => {
+  res.render("information.html");
 });
 
-app.get("/high-scores", (req,res)=> {
-    res.render("high-scores.html");
+app.post("/api/word", async (req, res) => {
+  const { numberOfLetters, duplicateLetters } = req.body;
+  const incorrectInputTypes =
+    isNaN(numberOfLetters) ||
+    numberOfLetters < 4 ||
+    numberOfLetters > 6 ||
+    typeof duplicateLetters !== "boolean";
+
+  if (incorrectInputTypes) {
+    res.status(422).json({ error: "Invalid data" });
+  }
+
+  try {
+    let words = fs.readFileSync("./server/words.json", "utf-8");
+    words = JSON.parse(words);
+    words = Object.keys(words);
+    const word = filterGameOptions(words, numberOfLetters, duplicateLetters);
+
+    const id = uuid.v4();
+    const newGame = new Game({
+      id: id,
+      startTime: new Date(),
+      guesses: [],
+      correctWord: word,
+      finished: false
+    });
+    await saveInDb(process.env.DB_URL, newGame);
```

Comment:
It should not be necessary to pass the URL around. It should be possible to just connect using mongoose in one place (i.e. right here in `app.js`) and then use `mongoose` without connecting again in other places.